include/aspect/material_model/averaging.h

```diff
@@ -0,0 +1,200 @@
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef __aspect__averaging_h
+#define __aspect__averaging_h
+
+#include <aspect/material_model/interface.h>
+#include <aspect/simulator_access.h>
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    using namespace dealii;
+    //namespace AveragingOperations
+    //		{
+    /**
+                         * An enum to define what kind of averaging operations are implemented.
+                         * These are:
+                         *
+                         * - No averaging, i.e., leave the values as they were provided by the
+                         * material model.
+                         *
+                         * - Arithmetic averaging: Set the values of each output quantity at
+                         * every quadrature point to \f[ \bar x = \frac 1Q \sum_{q=1}^Q x_q \f]
+                         * where $x_q$ are the values at the $Q$ quadrature points.
+                         *
+                         * - Harmonic averaging: Set the values of each output quantity at every
+                         * quadrature point to \f[ \bar x = \left(\frac 1Q \sum_{q=1}^Q
+                         * \frac{1}{x_q}\right)^{-1} \f] where $x_q$ are the values at the $Q$
+                         * quadrature points.
+                         *
+                         * - Geometric averaging: Set the values of each output quantity at
+                         * every quadrature point to \f[ \bar x = \left(\prod_{q=1}^Q
+                         * x_q\right)^{1/Q} \f] where $x_q$ are the values at the $Q$ quadrature
+                         * points.
+                         *
+                         * - Pick largest: Set the values of each output quantity at every
+                         * quadrature point to \f[ \bar x = \max_{1\le q\le Q} x_q \f] where
+                         * $x_q$ are the values at the $Q$ quadrature points.
+                         *
+                         * - Project to $Q_1$: This operation takes the values at the quadrature
+                         * points and computes the best bi- or trilinear approximation for them.
+                         * In other words, it projects the values into the $Q_1$ finite element
+                         * space. It then re-evaluate this projection at the quadrature points.
```

Comment:
The new NWS options are not documented here.

It's ok if you remove most of the text above and simply state that the first 5 values of the enum are as in MaterialModel::AveragingOperations and that the remainder are as explained in the documentation of the `averaging model' plugin.